### PR TITLE
[To rel/1.1][IOTDB-5719] Move DataNode's checking of SchemaQuota above the consensus layer

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

The current DataNode check for SchemaQuota is performed in the SchemaRegion. RatisConsensus expects the state machine to respond identically to the same apply request in the same state. However, in the case of multiple replicas, due to instabilities such as synchronization delays, the SchemaQuotaInfo may be different for different replicas, which may result in different SchemaRegion replicas responding differently to the same apply request (some replicas execute successfully, some replicas throw a QuotaExceededException).

Therefore, this PR moves the DataNode's checking of SchemaQuota above the consensus layer. The checks are performed before the execution plan enters the consensus layer to avoid errors.

The check cases involved are:
1. Creation of a time series
2. Creation of aligned time series
3. Activating metadata templates

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/schemaregion/SchemaRegionSchemaFileImpl.java [hunk 1]
```diff
@@ -790,6 +789,18 @@ public Map<Integer, MetadataException> checkMeasurementExistence(
...
   }
 
+  @Override
+  public void checkSchemaQuota(PartialPath devicePath, int timeSeriesNum)
+      throws SchemaQuotaExceededException {
+    if (schemaQuotaManager.getLevel().equals(ClusterSchemaQuotaLevel.TIMESERIES)) {
+      schemaQuotaManager.checkMeasurementLevel(timeSeriesNum);
+    } else if (schemaQuotaManager.getLevel().equals(ClusterSchemaQuotaLevel.DEVICE)) {
+      if (!mtree.checkDeviceNodeExists(devicePath)) {
+        schemaQuotaManager.checkDeviceLevel();
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/schemaregion/SchemaRegionSchemaFileImpl.java:799]: Is there any statistic variable representing device num that can be used in this judgement?

File: server/src/main/java/org/apache/iotdb/db/metadata/rescon/DataNodeSchemaQuotaManager.java [hunk 2]
```diff
@@ -40,6 +44,7 @@ public void updateRemain(long totalCount) {
   }
 
   public void checkMeasurementLevel(int acquireNumber) throws SchemaQuotaExceededException {
+    logger.info("checkMeasurementLevel {},{},{}", level.name(), limit, remain.get());
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/metadata/rescon/DataNodeSchemaQuotaManager.java:47]: It seems this log will be record each there's timeseries registration. Try reduce the log frequency.

**Review by MarcosZyk — COMMENTED** [review]
Please remember add device check when executing ```activateTemplate```, ```BatchActivateTemplate``` and ```InternalBatchActivateTemplate```.

**Comment by Cpaulyz** [comment]
> Please remember add device check when executing `activateTemplate`, `BatchActivateTemplate` and `InternalBatchActivateTemplate`.

done.

**Review by MarcosZyk — APPROVED** [review]

**Pull request merged**